File: src/main/java/by/devpav/serfor/domain/BasicEntity.java
package by.devpav.serfor.domain;

import javax.persistence.*;
import java.util.Objects;

@MappedSuperclass
public class BasicEntity {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected Long id;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof BasicEntity)) return false;
        BasicEntity that = (BasicEntity) o;
        return Objects.equals(getId(), that.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId());
    }
}

File: src/main/java/by/devpav/serfor/domain/BasicEntity.kt
package by.devpav.serfor.domain

import java.io.Serializable
import java.util.*
import javax.persistence.*

@MappedSuperclass
open class BasicEntity : Serializable {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is BasicEntity) return false
        val that = other as BasicEntity?
        return id == that!!.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }
}
